starknet_py/net/signer/stark_curve_signer.py

```diff
@@ -0,0 +1,67 @@
+from dataclasses import dataclass
+from typing import List
+
+from starkware.crypto.signature.signature import (
+    private_to_stark_key,
+)
+from starkware.starknet.core.os.transaction_hash.transaction_hash import (
+    calculate_transaction_hash_common,
+    TransactionHashPrefix,
+)
+
+from starknet_py.net.models import (
+    AddressRepresentation,
+    StarknetChainId,
+    Transaction,
+    parse_address,
+)
+from starknet_py.net.signer.base_signer import BaseSigner
+from starknet_py.utils.crypto.facade import message_signature
+
+
+@dataclass
+class KeyPair:
+    private_key: int
+    public_key: int
+
+    @staticmethod
+    def from_private_key(key: int) -> "KeyPair":
+        return KeyPair(private_key=key, public_key=private_to_stark_key(key))
+
+
+class StarkCurveSigner(BaseSigner):
+    def __init__(
+        self,
+        address: AddressRepresentation,
```

Comment:
Why is that part of the class and not the `sign_transaction` parameter?

Reply:
Signer is created once (or passed as a constructor) to AccountClient, so i don't think it makes much sense to have it as `sign_transaction` parameter.

Reply:
this address is account address right?

Reply:
if that's the case, we should name the parameter more clearly